controlplane/pkg/model/model.go

```diff
@@ -283,7 +284,10 @@ func (i *impl) AddDataplane(dataplane *Dataplane) {
 
 func (i *impl) DeleteDataplane(name string) {
 	i.Lock()
-	dataplane := i.dataplanes[name]
+	dataplane, ok := i.dataplanes[name]
+	if !ok {
+		return
+	}
 	delete(i.dataplanes, name)
 	i.Unlock()
```

Comment:
You need to convert this to a defer i.Unlock() further up, otherwise if the key isn't found, you will return before releasing the lock.